src/PowerShellEditorServices/Utility/Logging.cs

```diff
@@ -0,0 +1,234 @@
+            [CallerLineNumber] int callerLineNumber = 0);
+    }
+
+
+    /// <summary>
+    /// Manages logging and logger constructor for EditorServices.
+    /// </summary>
+    public static class Logging
+    {
+        /// <summary>
+        /// Builder class for configuring and creating logger instances.
+        /// </summary>
+        public class Builder
+        {
+            /// <summary>
+            /// The level at which to log.
+            /// </summary>
+            private LogLevel _logLevel;
+
+            /// <summary>
+            /// Paths at which to create log files.
+            /// </summary>
+            private Dictionary<string, LogLevel?> _filePaths;
+
+            /// <summary>
+            /// Whether or not to send logging to the console.
+            /// </summary>
+            private bool _useConsole;
+
+            /// <summary>
+            /// The log level to use when logging to the console.
+            /// </summary>
+            private LogLevel? _consoleLogLevel;
+
+            /// <summary>
+            /// Constructs An ILogger implementation builder instance with default configurations:
+            /// No log files, not logging to console, log level normal.
+            /// </summary>
+            public Builder()
+            {
+                _logLevel = Utility.LogLevel.Normal;
+                _filePaths = new Dictionary<string, LogLevel?>();
+                _useConsole = false;
+            }
+
+            /// <summary>
+            /// The severity level of the messages to log.
+            /// </summary>
+            /// <param name="logLevel">The severity level of the messages to log.</param>
+            /// <returns>the ILogger builder for reuse.</returns>
+            public Builder LogLevel(LogLevel logLevel)
+            {
+                _logLevel = logLevel;
+                return this;
+            }
+
+            /// <summary>
+            /// Add a path to output a log file to.
+            /// </summary>
+            /// <param name="filePath">The path ofethe file to log to.</param>
+            /// <param name="logLevel">The minimum log level for this file</param>
+            /// <param name="useMultiprocess">Set whether the log file should be readable by other processes</param>
+            /// <returns>the ILogger builder for reuse.</returns>
+            public Builder AddLogFile(string filePath, LogLevel? logLevel = null, bool useMultiprocess = false)
+            {
+                _filePaths.Add(filePath, logLevel);
+                return this;
+            }
+
+            /// <summary>
+            /// Configure the ILogger to send log messages to the console.
+            /// </summary>
+            /// <param name="logLevel">The minimum log level for console logging.</param>
+            /// <returns>the ILogger builder for reuse.</returns>
+            public Builder AddConsoleLogging(LogLevel? logLevel = null)
```

Comment:
Will this show up in the Integrated Terminal or in the VSCode logging view?

Reply:
I'm not sure. Possibly the latter if VSCode redirects stdout to its logging view.

I put it in so you can start EditorServices by itself and see logs printed to the console you start it in. I haven't actually tested that that works though.